content/src/content/jcr_root/apps/core/wcm/sandbox/components/carousel/v1/carousel/clientlibs/site/js/carousel.js

```diff
@@ -0,0 +1,268 @@
+                });
+            }
+
+            var next = that._elements["next"];
+            if (next) {
+                next.addEventListener("click", function() {
+                    if (that._active < that._elements["item"].length - 1) {
+                        that._active = that._active + 1;
+                    } else {
+                        that._active = 0;
+                    }
+                    refreshActive();
+                });
+            }
+
+            var indicators = that._elements["indicator"];
+            if (indicators) {
+                for (var i = 0; i < indicators.length; i++) {
+                    (function(index) {
+                        indicators[i].addEventListener("click", function(event) {
+                            navigate(index);
+                        });
+                    })(i);
+                }
+            }
+        }
+
+        /**
+         * Refreshes the item markup based on the current {@code Carousel#_active} index
+         *
+         * @private
+         */
+        function refreshActive() {
+            var items = that._elements["item"];
+            var indicators = that._elements["indicator"];
+
+            if (items) {
+                if (Array.isArray(items)) {
+                    for (var i = 0; i < items.length; i++) {
+                        if (i === parseInt(that._active)) {
+                            items[i].classList.add("cmp-carousel__item--active");
+                            items[i].removeAttribute("aria-hidden");
+                            indicators[i].classList.add("cmp-carousel__indicator--active");
+                            indicators[i].setAttribute("aria-selected", true);
+                        } else {
+                            items[i].classList.remove("cmp-carousel__item--active");
+                            items[i].setAttribute("aria-hidden", true);
+                            indicators[i].classList.remove("cmp-carousel__indicator--active");
+                            indicators[i].setAttribute("aria-selected", false);
+                        }
+                    }
+                } else {
+                    // only one item
+                    items.classList.add("cmp-carousel__item--active");
+                    indicators.classList.add("cmp-carousel__indicator--active");
+                }
+            }
+        }
+
+        /**
+         * Navigates to the item at the provided index
+         *
+         * @private
+         * @param {Number} index The index of the item to navigate to
+         */
+        function navigate(index) {
+            that._active = index;
+            refreshActive();
+        }
+    }
+
+    /**
+     * Reads options data from the Carousel wrapper element, defined via {@code data-cmp-*} data attributes
+     *
+     * @private
+     * @param {HTMLElement} element The Carousel element to read options data from
+     */
+    function readData(element) {
```

Comment:
I would call this method `getOptions`

Reply:
I would leave this as it is, it's consistent with the other UI widgets and should be centralized when we develop a widget component concept

Reply:
Works for me as well.